### DISCUSSION TITLE
Consider using generics for `Resize()`

### DISCUSSION THREAD

**Discussion by Raildex** [SixLabors/ImageSharp#2563] [Ideas]
The resampler argument of the mutator function `Resize` is a struct.
But due to the `IResampler` interface, the argument is boxed and stored on the heap.

![grafik](https://github.com/SixLabors/ImageSharp/assets/14938934/b2dcd80c-61d3-4123-9058-cae9ba5526bb)

**Comment by JimBobSquarePants** [comment]
I’m not sure what you are suggesting here. Is there some performance issue you have identified?

**Reply by Raildex** [reply]
I am creating mipmaps of large textures programmatically by using `Resize()` with selectable samplers in a batch.
The batch size can range from the thousands to ten thousands. The boxing and unboxing accumulates pretty fast in these cases.

**Reply by JimBobSquarePants** [reply]
Do you have some profiling output you could share? 

In your example code you should be using `KnownResamplers`.
